### PR TITLE
fix: Only reject approvals if subject has `endowment:caip25` permission revoked

### PR DESCRIPTION
**Pull request by FrederikBolding**
## **Description**

Rejecting all approvals caused strange behavior when installing Snaps from the Snaps Directory because the directory page uses `wallet_revokePermissions`. This PR fixes it by only rejecting pending approvals from the origin if the revoked permission is `endowment:caip25`.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/40139?quickstart=1)

## **Changelog**

CHANGELOG entry: Fixed an issue where approvals would be closed too soon

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/40112

## **Manual testing steps**

See attached issue

---

> [!NOTE]
> **Low Risk**
> Small conditional change in RPC middleware behavior with targeted test update; low risk aside from potential behavioral differences in edge cases around permission key selection.
> 
> **Overview**
> Updates `wallet_revokePermissions` so it only calls `rejectApprovalRequestsForOrigin` when the revoke set includes `endowment:caip25`, instead of rejecting approvals for any permission revocation.
> 
> Adjusts the corresponding unit test to trigger approval rejection via a CAIP-25-equivalent permission (`eth_accounts`) rather than an explicit `endowment:caip25` entry.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e5caae2835b6518dcf74b8b29fa7bde2786965d2. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: app/scripts/lib/rpc-method-middleware/handlers/wallet-revokePermissions.test.ts [hunk 1]
```diff
@@ -151,7 +151,7 @@ describe('revokePermissionsHandler', () => {
       ...baseRequest,
       params: [
         {
-          [Caip25EndowmentPermissionName]: {},
+          "eth_accounts": {},
```
Comment (FrederikBolding) [inline app/scripts/lib/rpc-method-middleware/handlers/wallet-revokePermissions.test.ts]: `endowment:caip25` is currently only revokable if specifying a legacy permission

**Review by scutuatua-crypto — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Review by jiexi — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
LGTM!
